libr/asm/arch/8051/8051_ass.c

```diff
@@ -413,22 +413,181 @@ static bool address_bit(char const* addr_str, ut8* addr_out) {
 	r_str_ncpy (bytepart, addr_str, separator - addr_str + 1);
 	bytepart[separator - addr_str + 1] = '\0';
 	r_str_ncpy (bitpart, separator + 1, strlen (separator));
-	if (!address_direct (bytepart, &byte)) {
-		goto end;
-	}
-	if (1 < strlen (bitpart)
-		|| bitpart[0] < '0' || '7' < bitpart[0]) {
-		ret = false;
-		goto end;
-	}
-	bit = bitpart[0] - '0';
-	if (0x20 <= byte && byte < 0x30) {
-		*addr_out = (byte - 0x20) * 8 + bit;
-		ret = true;
-	} else if (0x80 <= byte && !(byte%8)) {
-		*addr_out = byte + bit;
-		ret = true;
+
+	if (!strcmp (bytepart, "b")) {
+		if ((bitpart[0] - '0') <= 9) {
```

Comment:
the statement `if (bit <= 9)` appears in every block here. check it once before the elif chain instead:

```
if (bit <= 0) {
        ret = false;
        goto end;
}
```

and you can remove the `else {ret = false}` in every case.

Reply:
you can also set ret to true before the elif chain and set it to false in the final else to avoid having `ret = true` in every elif block.